sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -743,7 +928,8 @@ public String apply(String input) {
                           @ProcessElement
                           public void processElement(ProcessContext c) throws Exception {
                             String jobUuid = c.element();
-                            BigQuerySourceBase<T> source = createSource(jobUuid, coder);
+                            BigQuerySourceBase<T> source =
+                                (BigQuerySourceBase<T>) createSource(jobUuid, coder);
```

Comment:
Why are these casts needed? Maybe rather refactor `createSource` into two functions - one per method - and here call the one returning `BigQuerySourceBase`?

Reply:
Done.